controllers/core/openstackcontrolplane_controller.go

```diff
@@ -149,20 +149,30 @@ func (r *OpenStackControlPlaneReconciler) Reconcile(ctx context.Context, req ctr
 	//
 	if instance.Status.Conditions == nil {
```

Comment:
All the below condition initialization only runs if Conditions is nil. So it only run once. So if a service was first enabled then disabled then status of that service will not disappear from the condition list. It also means that if you disable a service which has a non Ready status, then OpenStackControlPlane will never reach Ready status. Is it OK? 
I think the current behavior of openstack-operator is that we stop reconciling a service when it is disabled but that also means we stop looking at the status of that service to incorporate it into the status of the OpenStackControlPlane. So if your goal here is to make OpenStackControlPlane Ready with disabled service then you might need to consider this case as well.

Reply:
> All the below condition initialization only runs if Conditions is nil. So it only run once. So if a service was first enabled then disabled then status of that service will not disappear from the condition list. It also means that if you disable a service which has a non Ready status, then OpenStackControlPlane will never reach Ready status. Is it OK? 

good spot, missed that

Reply:
This is a good point. Should probably add handling for that scenario. I'll look into that and push a new revision.